spark-screenshot-testing/src/test/kotlin/com/adevinta/spark/image/ImageScreenshot.kt

```diff
@@ -0,0 +1,266 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import coil3.annotation.ExperimentalCoilApi
+import coil3.asImage
+import coil3.compose.AsyncImageModelEqualityDelegate
+import coil3.compose.AsyncImagePainter
+import coil3.compose.AsyncImagePreviewHandler
+import coil3.compose.LocalAsyncImageModelEqualityDelegate
+import coil3.compose.LocalAsyncImagePreviewHandler
+import coil3.decode.DataSource
+import coil3.request.ErrorResult
+import coil3.request.ImageRequest
+import coil3.request.SuccessResult
+import com.adevinta.spark.DefaultTestDevices
+import com.adevinta.spark.InternalSparkApi
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.components.image.SparkImage
+import com.adevinta.spark.components.surface.Surface
+import com.adevinta.spark.components.text.Text
+import com.adevinta.spark.icons.SparkIcons
+import com.adevinta.spark.icons.Tattoo
+import com.adevinta.spark.paparazziRule
+import com.adevinta.spark.screenshot.testing.R
+import com.adevinta.spark.sparkSnapshot
+import com.android.ide.common.rendering.api.SessionParams.RenderingMode
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(InternalSparkApi::class)
+internal class ImageScreenshot {
+
+    private val imageSizes = listOf(48.dp, 100.dp, 200.dp)
+
+    @get:Rule
+    val paparazzi = paparazziRule(
+        deviceConfig = DefaultTestDevices.Tablet,
+        renderingMode = RenderingMode.H_SCROLL,
+    )
+
+    @Test
+    fun imagesInDifferentStates() {
+        paparazzi.sparkSnapshot {
+            ImageStates()
+        }
+    }
+
+    @Test
+    fun imageWithDifferentContentScales() {
+        paparazzi.sparkSnapshot {
+            ContentScaleShowcase()
+        }
+        // We don't use `sparkSnapshot` here since want the Image to be rendered instead of the loading state
+//        paparazzi.snapshot {
+//            SparkTheme {
+//                ContentScaleShowcase()
+//            }
+//        }
+    }
+
+    @OptIn(InternalSparkApi::class)
+    @Composable
+    private fun ImageStates() {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            modifier = Modifier.padding(16.dp),
+        ) {
+            Text(
+                text = "Image Component",
+                style = SparkTheme.typography.headline1,
+            )
+
+            val states = listOf("Empty", "Loading", "Error", "Success")
```

Comment:
Can we move this declaration to a local private enum class?
This would allow us to avoid hardcoding strings, and avoid future mistakes in the usage below.

Reply:
Sure, I'll look if we can reuse the one we have for the Configurator